File: leetcode/Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        
        
        if(s.size() == 0){
            return 0;
        }
        if(s.size() == 1){
            return 1;
        }
        int windows_size = 2;
        bool b = true;
        vector<char> l;
        string tmp_s;
        for(int i = 0; i<s.size(); ++i){
            b = true;
            if(i+windows_size > s.size()){
                break;
            }
            while(b){
                tmp_s.clear();
                //substr(a,b);   a是string index, 
                //b是指这个index后面多少个数，b不是index而是数量
                tmp_s = s.substr(i,windows_size);
                
                int before_size = tmp_s.size();
                
                vector<char> tmp_s_v(tmp_s.begin(), tmp_s.end());
                sort( tmp_s_v.begin(), tmp_s_v.end() );
                tmp_s_v.erase(unique( tmp_s_v.begin(), tmp_s_v.end() ), tmp_s_v.end() );
                int after_size = tmp_s_v.size();
                
                if (before_size != after_size){
                    b = false;
                } else{
                    windows_size++;
                }
                
                if(i+windows_size > s.size()){
                    break;
                }
                
                    
            }
            
        }
        
        return (windows_size-1);
    }
};

File: leetcode/Longest Substring Without Repeating Characters.py
# https://leetcode.com/problems/longest-substring-without-repeating-characters/description/
# Given a string, find the length of the longest substring without repeating characters.
# Input: "abcabcbb"
# Output: 3 
# Explanation: The answer is "abc", with the length of 3. 
class Solution:
    def lengthOfLongestSubstring(self,s):
        """
        :type s: str
        :rtype: int
        """
        l = []
        if len(s) == 0 :
            return 0
        if len(s) == 1:
            return 1
        windows_size = 2
        b = True
        for i in range(len(s)):
            b = True
            if i + windows_size > len(s):
                break      
            while(b):
                l.clear()
                tmp_s = s[i:i+windows_size]
                l = list(tmp_s)
                before_size = len(tmp_s)
                after_size = len(list(set(tmp_s)))
                if before_size != after_size:
                    b = False
                else:
                    windows_size += 1
                if i + windows_size > len(s):
                    break
        return windows_size-1
